java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -364,6 +415,31 @@ public static final class Builder {
     private boolean useSystemTrustStore;
     private BufferAllocator allocator;
 
+    public Builder() {
+
+    }
+
+    /**
+     * Copies the builder.
+     *
+     * @param original The builder to base this copy off of.
+     */
+    private Builder(Builder original) {
+      this.middlewareFactories.addAll(original.middlewareFactories);
```

Comment:
I just want to verify, this would ensure the CookiesMiddleware gets copied over for the Endpoint clients, correct?

Reply:
Ah, I didn't think about that, but yes, it should mean that the middleware gets shared between the clients. That might be good or bad depending on the server implementation but we can follow up if it turns out to be an issue.

Reply:
A new CookieMiddlewareFactory is created for subsequent connections (same with the AuthMiddlewareFactory). This was intentional but should actually be configurable.

Reply:
Added #38576 . This one can cause breakage for existing users that are relying on the same cookie/auth token from being sent so I'll pick it up next.

Reply:
Awesome thanks! we depend on cookies to maintain state and expect the callOptions we set to go out with every request so that will be important for us!